controller/readwrite.go

```diff
@@ -145,27 +145,28 @@ func (rw *ReadWrite) checkApply(ctx context.Context, u unit) (bool, error) {
 			taskName, err)
 	}
 
-	ev, err := event.NewEvent(taskName, &event.Config{
-		Providers: u.providers,
-		Services:  u.services,
-		Source:    u.source,
-	})
-	if err != nil {
-		return false, fmt.Errorf("error creating event for task %s: %s",
-			taskName, err)
-	}
-	defer func() {
-		ev.End(err)
-		log.Printf("[TRACE] (ctrl) adding event %s", ev.GoString())
-		err = rw.store.Add(*ev)
-	}()
-
 	// result.Complete is only `true` if the template has new data that has been
 	// completely fetched. Rendering a template for the first time may take several
 	// cycles to load all the dependencies asynchronously.
 	if result.Complete {
 		log.Printf("[DEBUG] (ctrl) change detected for task %s", taskName)
+
+		ev, err := event.NewEvent(taskName, &event.Config{
+			Providers: u.providers,
+			Services:  u.services,
+			Source:    u.source,
+		})
+		if err != nil {
+			return false, fmt.Errorf("error creating event for task %s: %s",
+				taskName, err)
+		}
 		ev.Start()
+		defer func() {
+			ev.End(err)
```

Comment:
Can you change the variable name of `err` to be more identifiable? This seems to not  include `resolver.Run()` which also returns `err` but the variable is reinitialized from `NewEvent`.

Should errors from fetching template dependencies be included in event tracking?

Reply:
I like the idea of naming the stored error something different! I'll do that.

Re errors from fetching template dependencies: 🤔 I wasn't considering them for event tracking since there's multiple fetchings per task-execution, and I was worried that it would skew the status data. Thinking about this again, to your point, I think that it would be good to include the errors but not the successes. A skew in successful events might wash out failed task executions but a skew in failed events isn't as problematic (since the task can't even execute). Thanks for asking! Let me make this change and let me know if you have thoughts.